spec/models/service_spec.rb

```diff
@@ -126,4 +126,21 @@
       expect(service.token).to eq('agithubtoken')
     end
   end
+
+  describe 'omniauth options provider registry for non-conforming omniauth responses' do
+    describe '.register_options_provider' do
+      it 'allows gem developers to add their own options provider to the registry' do
+        Service.register_options_provider('test-omniauth-provider') do |omniauth|
+          { name: omniauth['special_field'] }
+        end
+
+        actual_options = Service.get_options({
+          'provider' => 'test-omniauth-provider',
+          'special_field' => 'A Great Name'
+        })
+
+        expect(actual_options[:name]).to eq('A Great Name')
+      end
```

Comment:
I think from the application perspective it is the right way to always have them global, for the tests i don't worry much about it becoming a bit "ugly".